<launch>
  <arg name="input/image" default="~/image_raw"/>
  <arg name="input/rois" default="~/rois"/>
  <arg name="output/traffic_signals" default="classified/traffic_signals"/>

  <arg name="use_gpu" default="true"/>
  <!-- classifier_type {hsv_filter: 0, cnn: 1} -->
  <arg name="classifier_type" default="1" if="$(var use_gpu)"/>
  <arg name="classifier_type" default="0" unless="$(var use_gpu)"/>

  <arg name="build_only" default="false" description="shutdown node after TensorRT engine file is built"/>

  <node pkg="traffic_light_classifier" exec="traffic_light_classifier_node" name="traffic_light_classifier">
    <remap from="~/input/image" to="$(var input/image)"/>
    <remap from="~/input/rois" to="$(var input/rois)"/>
    <remap from="~/output/traffic_signals" to="$(var output/traffic_signals)"/>
    <param name="approximate_sync" value="false"/>
    <param name="classifier_type" value="$(var classifier_type)"/>
    <param name="classifier_model_path" value="$(find-pkg-share traffic_light_classifier)/data/traffic_light_classifier_efficientNet_b1.onnx"/>
    <param name="classifier_label_path" value="$(find-pkg-share traffic_light_classifier)/data/lamp_labels.txt"/>
    <param name="classifier_precision" value="fp16"/>
    <param name="build_only" value="$(var build_only)"/>
  </node>
</launch>
